.github/workflows/deno-cd.yml

```diff
@@ -0,0 +1,44 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Deno Continuous Deployment
+
+on:
+  push:
+    paths:
+      - 'src/version.ts'
+    branches: [main]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v2
+
+      - name: Setup Deno
+        # uses: denoland/setup-deno@v1
+        uses: denoland/setup-deno@004814556e37c54a2f6e31384c9e18e983317366
+        with:
+          deno-version: v1.x
+
+      - name: Extract Tag Name from version.ts
+        id: save_version
+        run: |
+          echo "TAG_NAME=$(deno run -q --allow-read src/version.ts)" >> $GITHUB_ENV
+          echo "We will now create the ${{ env.TAG_NAME }} tag."
+
+      - name: Create and push git tag
+        uses: rickstaa/action-create-tag@v1
+        with:
+          tag: "${{ env.TAG_NAME }}"
```

Comment:
Do we know what happens if this errors? I.e. the tag is not new or in the wrong format

Reply:
The [action-create-tag](https://github.com/rickstaa/action-create-tag) action is set up to error out [if any of its commands error out](https://github.com/rickstaa/action-create-tag/blob/main/entrypoint.sh#L2). The action basically boils down to a `git tag X` and `git push origin X`, so if either of those commands fail, the GitHub workflow would fail. We can optionally set this action to [force push a tag](https://github.com/rickstaa/action-create-tag#force_push_tag) to avoid a 'tag already exists' error but I don't think we want that. I think failure is good in this situation as that would signify that some preconditions are not met (tag already exists, tag name is wrong or invalid), and we would want the workflow to fail.